### PR TITLE
add bombman npc

### PR DESCRIPTION
**Pull request by Alx-Lai**
as title

### CODE REVIEW

File: extensions/bombman/standalone.mjs [hunk 1]
```diff
@@ -41,6 +41,7 @@ class Standalone {
     // set dynamic cell set: bombmanHasBomb
     this.bombCells = new Map(); // key: a unique bomb ID; value: the cell
     this.bombID = 0;
+    this.gameStart = false;
```
Comment (ktpss95112) [inline extensions/bombman/standalone.mjs:44]: `this.gameStarted`

File: extensions/bombman/standalone.mjs [hunk 2]
```diff
@@ -140,10 +141,12 @@ class Standalone {
     // mapCoord has to be inside an arena
     // TODO: use the utility function in maplib if there is such function
     const inside = this.insideMap(mapCoord);
-    if (!inside) return;
+    if (!inside || !this.gameStart) return false;
+    // ghost mode no attack
+    if (this.helper.gameState.getPlayer(player.playerID).ghostMode) return false;
```
Comment (ktpss95112) [inline extensions/bombman/standalone.mjs:146]: As we have just discussed, maybe we can ignore this check.

File: extensions/bombman/standalone.mjs [hunk 3]
```diff
@@ -240,6 +243,80 @@ class Standalone {
...
+   * @param {*} srcExt
+   * @param {String} playerID
+   * @param {Object} kwargs
+   * @param {*} sfInfo
+   * @return {String} - the next state
+   */
+  async s2s_sf_startBombman(srcExt, playerID, kwargs, sfInfo) {
+    const {next} = kwargs;
+    if (this.gameStart) {
+      console.log('game have already started');
```
Comment (ktpss95112) [inline extensions/bombman/standalone.mjs:258]: Consider more descriptive logging such as `console.log('[bombman] the game have already started');`

File: extensions/bombman/standalone.mjs [hunk 4]
```diff
@@ -240,6 +243,80 @@ class Standalone {
...
+   * @param {*} srcExt
+   * @param {String} playerID
+   * @param {Object} kwargs
+   * @param {*} sfInfo
+   * @return {String} - the next state
+   */
+  async s2s_sf_resetBombman(srcExt, playerID, kwargs, sfInfo) {
+    const {next} = kwargs;
+    if (!this.gameStart) {
+      console.log('The game has not started yet');
```
Comment (ktpss95112) [inline extensions/bombman/standalone.mjs:276]: more descriptive logging

File: run/npc/bombman-npc.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            "func": "resetBombman",
+            "kwargs": {
+                "next": "s4"
+            }
+        },
+        "s4": {
+            "func": "exit",
+            "kwargs": {
+                "next": "s1"
+            }
```
Comment (ktpss95112) [inline run/npc/bombman-npc.json:49]: mixed tab size

**Review by ktpss95112 — COMMENTED** [review]
as the comments show

**Review by ktpss95112 — APPROVED** [review]

**Review by john0312 — APPROVED** [review]

**Pull request merged**